priv/repo/migrations/20230713132951_add_extra_to_assets.exs

```diff
@@ -0,0 +1,9 @@
+defmodule Beacon.Repo.Migrations.AddExtraToAssets do
+  use Ecto.Migration
+
+  def change do
+    alter table(:beacon_assets) do
+      add :extra, :map, default: %{}
```

Comment:
We can, but in DB land it's nice to have that where `extra IS NOT NULL` clause instead of having to sort out some  SQL for empty json objects.

Reply:
Either way we'll need a query `where extra = '{}'` to find empty objects because the default is `%{}` which means we'll end with `nil` and `%{}` representing "there's no extra data for this asset", right? So maybe drop the `default: %{}` all together and allow only either `nil` or non-empty maps `%{"k"=>"v"}`?

Reply:
oh, you're right. That went right past me.